app/services/sendgrid_api.rb

```diff
@@ -0,0 +1,56 @@
+class SendgridApi
+  extend SingleForwardable
+
+  def_single_delegators :new, :spam_reported?, :bounced?, :smtp_alive?
+
+  def spam_reported?(email)
```

Comment:
:thought_balloon: Perhaps we could DRY the predicate methods up a bit.. however I am wondering if these two methods should be classed as just coincidentally similar for the time being and it might be a bit naive to extract a generalised helper method. If not perhaps something like this:

```ruby
class SendgridApi
  def spam_reported?(email)
    such_api { spam_reports.retrieve(email: email).any? }
  end

  def bounced?(email)
    such_api { bounces.retrieve(email: email).any? }
  end

  private

  def such_api
    return false unless can_access_sendgrid?
    yield
  rescue JSON::ParserError, SendgridToolkit::APIError
    false
  end
end
```